src/app/components/layout/header/nav-bar/nav-bar.component.spec.ts

```diff
@@ -1,7 +1,14 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { Pipe, PipeTransform } from '@angular/core';
+import { MatIconModule, MatProgressSpinnerModule, MatTooltip } from '@angular/material';
 
 import { NavBarComponent } from './nav-bar.component';
+import { NavBarService } from '../../../../services/nav-bar.service';
+import { DoubleButtonComponent } from '../../double-button/double-button.component';
+import { ButtonComponent } from '../../button/button.component';
```

Comment:
Yes, when "DoubleButtonComponent" is added to the to declarations array, it requires all inner components to be declared as well. Otherwise there is the "Can't bind to 'disabled' since it isn't a known property of 'app-button'." error and it lasts till the last inner component included.